addons/stock/wizard/product_replenish.py

```diff
@@ -38,10 +38,12 @@ def default_get(self, fields):
                 product_tmpl_id = product_id.product_tmpl_id
                 res['product_tmpl_id'] = product_id.product_tmpl_id.id
                 res['product_id'] = product_id.id
+                res['quantity'] = abs(product_id.virtual_available) if product_id.virtual_available < 0 else 1
             elif self.env.context.get('default_product_tmpl_id'):
                 product_tmpl_id = self.env['product.template'].browse(self.env.context['default_product_tmpl_id'])
                 res['product_tmpl_id'] = product_tmpl_id.id
                 res['product_id'] = product_tmpl_id.product_variant_id.id
+                res['quantity'] = abs(product_tmpl_id.virtual_available) if product_tmpl_id.virtual_available < 0 else 1
```

Comment:
I think it should be better to set the `quantity` in a onchange of `product_id` because if the product template has variants, it doesn't make a lot of sense to set the quantity equal to the product template forecasted quantity.
Do you agree @thomasdeleval ?